File: numguess/numguess.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().Unix())
	actualNum := rand.Intn(10000000)
	tries := 0
	for {
		guessNum := rand.Intn(10000000)
		if guess(guessNum, actualNum) == 3 {
			fmt.Printf("The number was %d\n", guessNum)
			break
		}
		tries++
	}
	fmt.Printf("It took %d tries\n", tries)

}

func guess(guessNum, actualNum int) int {
	if guessNum != actualNum {
		if guessNum > actualNum {
			return 1
		}
		if guessNum < actualNum {
			return 2
		}
	} else {
		return 3
	}
	return 0
}

File: numguess/numguess.c
#include <time.h>
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>


bool numguess(int guessNum, int actualNum){
    if(guessNum<actualNum){
        return false;
    }else if (guessNum>actualNum) {
        return false;
    }else{
        return true;
    }
}
int main(){
    int actualNum, guessNum, Attempts;
    int tries = 0;

    srand(time(NULL));
    actualNum = rand()%10000000;
    while(true){
        guessNum = (rand()%10000000)+1;
        if (numguess(guessNum, actualNum)){
            printf("The number was: %d\n", guessNum);
            break;
        }
        tries++;
    }
    printf("It took %d tries\n", tries);
    return 0;
}
